subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/GradleBuildInstantExecutionSmokeTest.groovy

```diff
@@ -78,7 +78,7 @@ class GradleBuildInstantExecutionSmokeTest extends AbstractSmokeTest {
 
         then:
         result.output.count("Reusing instant execution cache") == 1
-        result.task(":distributions:binZip").outcome == TaskOutcome.UP_TO_DATE
+        result.task(":distributions:binZip").outcome == TaskOutcome.SUCCESS
```

Comment:
What caused this to be out-of-date? We put a lot of effort into making that one line pass and we should try hard to keep it passing. This is one of the most important assertions in all of the instant execution test coverage.

Reply:
@adammurdoch it seems honouring the `mustRunAfter` constraints made the instant execution build match the behaviour of the vintage build:
```
Task ':distributions:binZip' is not up-to-date because:
  Output property 'archiveFile' file ./build/distributions/gradle-6.5-bin.zip has been removed.
```

Reply:
The smoke test reverted to the previous behaviour after I made the PR more conservative by only serializing the dependency and must successor nodes of each scheduled node.

That means the change in behaviour was due to the serialization of the finalizing successors.

Reply:
@adammurdoch mistery solved! https://github.com/gradle/gradle/pull/13072